### PR TITLE
UI - Add ability to restart emulation.

### PR DESCRIPTION
**Pull request by saldabain**
Fixes https://github.com/Ryujinx/Ryujinx/issues/2974

Adds a menu item to restart emulation of the current game.

Needs review and testing. May not work with games that launch games inside of them. (SMAS)

Re-created the PR since the previous was ruined w/ a bad rebase.

### CODE REVIEW

File: Ryujinx/Ui/MainWindow.cs [hunk 1]
```diff
@@ -1168,7 +1179,7 @@ private void Row_Activated(object sender, RowActivatedArgs args)
 
             string path = (string)_tableStore.GetValue(treeIter, 9);
 
-            LoadApplication(path);
+            LoadApplication(path, /*startFullscreen=*/false, /*isUserSelectedPath=*/true);
```
Comment (gdkchan) [inline Ryujinx/Ui/MainWindow.cs:1182]:
```suggestion
            LoadApplication(path, startFullscreen: false, isUserSelectedPath: true);
```

Comment (gdkchan) [reply Ryujinx/Ui/MainWindow.cs:1182]: Same for the other ones.

File: Ryujinx/Ui/MainWindow.cs [hunk 2]
```diff
@@ -1318,6 +1329,15 @@ private void Window_Close(object sender, DeleteEventArgs args)
             }
         }
 
+        private void RestartEmulation_Pressed(object sender, EventArgs args)
+        {
+            if (_emulationContext != null)
+            {
+                _userChannelPersistence.ExecuteProgram(HLE.HOS.Services.Am.AppletOE.ApplicationProxyService.ApplicationProxy.Types.ProgramSpecifyKind.RestartProgram, 0);
```
Comment (gdkchan) [inline Ryujinx/Ui/MainWindow.cs:1336]: Add a `using` instead of this super long line.

File: Ryujinx/Ui/MainWindow.cs (hunk 1 above)
Comment (saldabain) [reply Ryujinx/Ui/MainWindow.cs]: done

File: Ryujinx/Ui/MainWindow.cs (hunk 2 above)
Comment (saldabain) [reply Ryujinx/Ui/MainWindow.cs]: done

**Review by SiongSng — APPROVED** [review]

**Comment by MutantAura** [comment]
Works great across all my games other than as you mention SM3DAS. Restart works for each individual title, for instance playing Galaxy will restart _just_ Galaxy which I guess is actually good behaviour but restarting on the selection screen will hang.

No clue why anyone would want to restart the game selection though so I doubt this is a massive issue.

**Review by riperiperi — APPROVED** [review]

**Comment by riperiperi** [comment]
Looks good. In future to prevent lost reviews and comment history just force push changes to your branch.

**Review by Shahil-Ayato — APPROVED** [review]

**Review by Niwu34 — APPROVED** [review]

**Review by GamerzHell9137 — APPROVED** [review]

**Comment by TSRBerry** [comment]
This needs a rebase.

**Review by marysaka — COMMENTED** [review]
Need a reabse and implementation on Avalonia side.

**Pull request closed without merging**